### PR TITLE
Add S3 security mapping

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

**Comment by tooptoop4** [comment]
currently don't have to specify any role name, can that behaviour ( ie InstanceProfileCredentialsProvider) be the default if no json provided?

**Comment by electrum** [comment]
@tooptoop4 I'm not sure I understand the question. The new security mapping support should not change any existing behavior -- if you don't enable it, things will work the same as before. I'm also happy to chat on Slack if that's easier.

**Comment by tooptoop4** [comment]
@electrum "com.amazonaws.auth.InstanceProfileCredentialsProvider" imports were removed ?

**Comment by electrum** [comment]
Yes, see the description for the **Remove config hive.s3.use-instance-credentials** commit. Instance credentials are still supported via `DefaultAWSCredentialsProviderChain`, which is why that property is not needed.

File: presto-docs/src/main/sphinx/connector/hive.rst [hunk 1]
```diff
@@ -429,6 +429,101 @@ or credentials for a specific use case (e.g., bucket/user specific credentials).
...
+          "allowedIamRoles": [
+            "arn:aws:iam::123456789101:role/test1",
+            "arn:aws:iam::123456789101:role/test2",
+            "arn:aws:iam::123456789101:role/test3"
+          ]
+        },
+        {
+          "prefix": "s3://special-bucket/",
+          "accessKey": "AKIAxxxaccess",
+          "secretKey": "iXbXxxxsecret"
```
Comment (kokosing) [inline presto-docs/src/main/sphinx/connector/hive.rst]: IIRC there was an issue with specifing `accessKey` and `secretKey` together `iamRole`. Should we document this? Or maybe even disallow such mappings in the code.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/s3/S3SecurityMappingConfig.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    private String roleCredentialName;
+    private Duration refreshPeriod;
+    private String colonEscape;
+
+    public Optional<File> getConfigFile()
+    {
+        return Optional.ofNullable(configFile);
+    }
+
+    @Config("hive.s3.security-mapping.config-file")
```
Comment (kokosing) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/s3/S3SecurityMappingConfig.java]: Add `@ConfigDescription` to these.

File: presto-docs/src/main/sphinx/connector/hive.rst (hunk 1 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/connector/hive.rst]: You're right, the current code in `PrestoS3FileSystem#createAwsCredentialsProvider` returns `AWSStaticCredentialsProvider` if credentials exist. However, the intent is that they can be used together, so we should fix that to make these the `withLongLivedCredentials(...)` for `STSAssumeRoleSessionCredentialsProvider`.

**Pull request merged**